pmlc/dialect/tile/program.h

```diff
@@ -19,11 +20,17 @@ struct ProgramArgument {
   pmlc::util::BufferPtr buffer;
 };
 
+struct ProgramDType {
+  mlir::FloatType floatx;
+  mlir::IntegerType intx;
+};
+
```

Comment:
If we kept the constant materialization pass local to the builder, we wouldn't need to expose this outwards to the later compiler/execution API.